File: patterns/notifications-in-groups/rules-js/notifications-in-groups.js
rules.JSRule({
  name: "Warnings and alerts for temperature",
  triggers: [triggers.GroupStateChangeTrigger("TemperatureRooms")],
  execute: (event) => {
    const hour = time.ZonedDateTime.now().hour();
    const members = items.getItem("TemperatureRooms").members;

    if (hour >= 9 && hour < 21) { // 9 bis 21 Uhr
      members
        .filter((t) => parseFloat(t.state) >= 25 && parseFloat(t.state) < 30)
        .forEach((r) => {
          console.log("Temp warn " + r.name + ": " + r.state + " Grad C");
          // Weitere Benachrichtigungen
        });
    }

    members
      .filter((t) => parseFloat(t.state) >= 30)
      .forEach((r) => {
        console.log("Temp alert " + r.name + ": " + r.state + " Grad C");
        // Weitere Benachrichtigungen
      });
  }
});

File: patterns/notifications-in-groups/rules-python/notifications_in_groups.py
from datetime import datetime

from openhab import rule, Registry
from openhab.triggers import GroupStateChangeTrigger


@rule(triggers=[GroupStateChangeTrigger("TemperatureRooms")])
class WarningsAndAlertsForTemperature:
    def execute(self, module, input):
        hour = datetime.now().hour
        members = Registry.getItem("TemperatureRooms").getAllMembers()

        if 9 <= hour < 21:  # 9 bis 21 Uhr
            for r in members:
                state = str(r.getState())
                if state in ("NULL", "UNDEF"):
                    continue
                temp = float(state)
                if 25 <= temp < 30:
                    self.logger.info("Temp warn {}: {} Grad C".format(r.getName(), temp))
                    # Weitere Benachrichtigungen

        for r in members:
            state = str(r.getState())
            if state in ("NULL", "UNDEF"):
                continue
            temp = float(state)
            if temp >= 30:
                self.logger.info("Temp alert {}: {} Grad C".format(r.getName(), temp))
                # Weitere Benachrichtigungen
